patches/minecraft/net/minecraft/inventory/Container.java.patch

```diff
@@ -1,6 +1,25 @@
 --- ../src-base/minecraft/net/minecraft/inventory/Container.java
 +++ ../src-work/minecraft/net/minecraft/inventory/Container.java
-@@ -650,7 +650,7 @@
+@@ -76,12 +76,16 @@
+ 
+             if (!ItemStack.func_77989_b(itemstack1, itemstack))
+             {
++                boolean nonCapDifference = itemstack == null || itemstack1 == null || (!itemstack.hasCapabilities() && !itemstack1.hasCapabilities()) || !ItemStack.areItemStacksEqualIgnoreCaps(itemstack1, itemstack);
+                 itemstack1 = itemstack == null ? null : itemstack.func_77946_l();
+                 this.field_75153_a.set(i, itemstack1);
+ 
+-                for (int j = 0; j < this.field_75149_d.size(); ++j)
++                if (nonCapDifference)
+                 {
+-                    ((IContainerListener)this.field_75149_d.get(j)).func_71111_a(this, i, itemstack1);
++                    for (int j = 0; j < this.field_75149_d.size(); ++j)
++                    {
++                        ((IContainerListener)this.field_75149_d.get(j)).func_71111_a(this, i, itemstack1);
++                    }
```

Comment:
Remove the indentation here and just place `if (nonCapDifference)` above the for statement and this whole crap becomes a 1 line addition patch. You can then also inline the variable definition from above and that goes away too.